src/lib/containers/DiscussionSearchResult.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState, useEffect } from 'react'
+import { getReply, getReplyMessageUrl, getThreadMessageUrl, getThread, getUserProfileById } from '../utils/SynapseClient'
+import { useSynapseContext } from '../utils/SynapseContext'
+import { UserProfile } from '../utils/synapseTypes'
+import { DiscussionReplyBundle, DiscussionThreadBundle } from '../utils/synapseTypes/DiscussionBundle'
+import moment from 'moment'
+import ReplyIcon from '../assets/icons/reply.svg'
+import ThreadIcon from '../assets/icons/thread.svg'
+import Typography from '../utils/typography/Typography'
+import { Col, Row } from 'react-bootstrap'
+import UserCard from './UserCard'
+import { SMALL_USER_CARD } from '../utils/SynapseConstants'
+
+export const getMessage = async (url: string): Promise<any> => {
+    const response = await fetch(url, {
+        method: 'GET',
+        headers: {
+            Accept: '*/*',
+            'Access-Control-Request-Headers': 'authorization',
+            'Content-Type': 'text/plain; charset=utf-8'
+        }
+    })
+    return response.text()
+}
+
+export type DiscussionSearchResultProps = {
+    threadId: string,
+    replyId?: string,
+}
+
+const DiscussionSearchResult = (props: DiscussionSearchResultProps) => {
+    const { threadId, replyId } = props
+    const { accessToken } = useSynapseContext()
+    const [threadBundle, setThreadBundle] = useState<DiscussionThreadBundle>()
+    const [messageUrl, setMessageUrl] = useState<string>('')
+    const [replyBundle, setReplyBundle] = useState<DiscussionReplyBundle>()
+    const [replyAuthor, setReplyAuthor] = useState<UserProfile>()
+
+    const getThreadOrReply = async () => {
+        let someUrl
+        const thread = await getThread(threadId, accessToken)
+        if (replyId) {
+            const reply = await getReply(replyId, accessToken)
+            someUrl = await getReplyMessageUrl(reply.messageKey, accessToken)
+            setReplyAuthor(await getUserProfileById(accessToken, reply.createdBy))
+            setReplyBundle(reply)
+        } else {
+            setReplyAuthor(await getUserProfileById(accessToken, thread.createdBy))
+            someUrl = await getThreadMessageUrl(thread.messageKey, accessToken)
+        }
+        setMessageUrl(await getMessage(someUrl.messageUrl))
+        setThreadBundle(thread)
+    }
+
+    useEffect(() => {
+        getThreadOrReply()
+    }, [])
+
+    const getUrl = (threadId: string, replyId: string, projectId: string) => {
```

Comment:
I think replyId should be optional (and only set if the replyId is set in the search result), right?  If unset, it would be better to not include it in the URL query param in this URL.